### PR TITLE
users service with sqlite database connection

### PR DESCRIPTION
**Pull request by rolcho**

### CODE REVIEW

File: sk0-back/src/users/dtos/create-user.dto.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export class CreateUserDto {
+  @IsEmail()
+  email: string;
+
+  @IsStrongPassword(passwordRules)
+  password: string;
+
+  @IsString()
+  name: string;
+}
```
Comment (markkovari) [inline sk0-back/src/users/dtos/create-user.dto.ts]: Maybe we should add a minimum length here? Still not sure about the use cases. (Might definitely happen later)

File: sk0-back/src/users/services/users.service.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { CreateUserDto } from '../dtos/create-user.dto';
+
+@Injectable()
+export class UsersService {
+  constructor(@Inject(USER_REPOSITORY) private repo: Repository<User>) {}
+
+  create(user: CreateUserDto): Promise<User> {
+    const createdUser = this.repo.create({ ...user });
+    return this.repo.save(createdUser);
+  }
```
Comment (markkovari) [inline sk0-back/src/users/services/users.service.ts]: I would explicitly say that the functions (or methods) are async in this class, (for `create` and `findAll`) I know that you did not use the await keyword in these methods, but would say that these are async operations and not just inferred.

File: sk0-back/src/users/user.entity.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity()
+export class User {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  email: string;
```
Comment (markkovari) [inline sk0-back/src/users/user.entity.ts]: Not sure about the registration scenarios, but I think email should be unique.

File: sk0-back/src/constants.ts [hunk 4]
```diff
@@ -1 +1,2 @@
 export const DATA_SOURCE = 'DataSource';
+export const USER_REPOSITORY = 'UserRepository';
```
Comment (markkovari) [inline sk0-back/src/constants.ts]: We might want to use `as const` for constant value declarations.

```suggestion
export const DATA_SOURCE = 'DataSource' as const;
export const USER_REPOSITORY = 'UserRepository' as const;
```

or we can use Symbols rather.

**Review by markkovari — COMMENTED** [review]
Looking good, already bunch of tests. 💪🏿

File: sk0-back/src/users/dtos/create-user.dto.ts (hunk 1 above)
Comment (rolcho) [reply sk0-back/src/users/dtos/create-user.dto.ts]: Thx changed for 3 now

File: sk0-back/src/constants.ts (hunk 4 above)
Comment (rolcho) [reply sk0-back/src/constants.ts]: using as const.

File: sk0-back/src/users/services/users.service.ts (hunk 2 above)
Comment (rolcho) [reply sk0-back/src/users/services/users.service.ts]: made some refactoring

File: sk0-back/src/users/user.entity.ts (hunk 3 above)
Comment (rolcho) [reply sk0-back/src/users/user.entity.ts]: yes you are right, made the change, thanks

**Comment by rolcho** [comment]
> Looking good, already bunch of tests. 💪🏿

I ❤️ testing

**Review by bendesbeus — APPROVED** [review]

**Review by Jankita58 — APPROVED** [review]

**Review by Thomas851204 — APPROVED** [review]
Majd az env setup-ról és a provider részről kapunk egy kis felvilágosítást? :)

**Pull request merged**